pixeltable/exprs/type_cast.py

```diff
@@ -0,0 +1,68 @@
+import json
+from typing import Optional, Dict, List, Tuple, Any
+
+import sqlalchemy as sql
+
+from pixeltable.exprs import DataRow, Expr, RowBuilder
+from pixeltable.type_system import ColumnType
+
+
+class TypeCast(Expr):
+
+    def __init__(self, underlying: Expr, new_type: ColumnType):
+        super().__init__(new_type)
+        if self.col_type.is_string_type() \
+                or self.col_type.is_json_type() and underlying.col_type.is_string_type():
+            # It's a valid type conversion
+            self.components = [underlying]
+            self.id = self._create_id
+        else:
+            raise RuntimeError(f'Expression of type `{self._underlying.col_type}` cannot be cast to `{self.col_type}`')
+
+    @property
+    def _underlying(self):
+        return self.components[0]
+
+    def _equals(self, other: Expr) -> bool:
+        if isinstance(other, TypeCast):
```

Comment:
This check isn't necessary (see Expr.equals()).

Reply:
ah, thanks. I misunderstood the pattern. `return True` should suffice here.